.github/workflows/TauDEM.yaml

```diff
@@ -12,65 +12,46 @@ concurrency:
 
 jobs:
   TauDEM:
-    runs-on: ${{ matrix.config.os }}
+    runs-on: windows-latest
 
-    name: ${{ matrix.config.os }} (${{ matrix.config.r }})
-
-    strategy:
-      fail-fast: false
-      matrix:
-        config:
-          - {os: macOS-latest,   r: 'release'}
-          - {os: windows-latest, r: 'release'}
-          - {os: ubuntu-latest,   r: 'release'}
+    name: windows-release
 
     env:
       GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
       R_KEEP_PKG_SOURCE: yes
 
     steps:
       - uses: actions/checkout@v3
-
-      - name: Install required pkgs for linux
-        if: runner.os == 'Linux'
-        run: |
-          sudo apt-get update
-          sudo apt-get install gdal-bin libgdal-dev mpich
-
-      - name: Install required pkgs for mac
-        if: runner.os == 'macOS'
-        run: |
-          brew install gdal mpich
-
-      - name: Install TauDEM on linux and mac
-        if: runner.os != 'Windows'
-        run: sh ./TauDEM.sh
-
+      
+      - name: Setup MPI
+        uses: mpi4py/setup-mpi@v1
```

Comment:
It's not a "popular" action but I wonder if using it is not easier for maintenance.
With the former commands we had, to update the download link I had to use the requests section of the web dev console, so it's not that straightforward.